src/core/thread/mle.cpp

```diff
@@ -295,10 +296,16 @@ ThreadError Mle::Restore(void)
 
     if (networkInfo.mDeviceState == kDeviceStateChild)
     {
-        length = sizeof(mParent);
+        length = sizeof(parentInfo);
         SuccessOrExit(error = otPlatSettingsGet(mNetif.GetInstance(), kKeyParentInfo, 0,
-                                                reinterpret_cast<uint8_t *>(&mParent), &length));
-        VerifyOrExit(length == sizeof(mParent), error = kThreadError_NotFound);
+                                                reinterpret_cast<uint8_t *>(&parentInfo), &length));
+        VerifyOrExit(length == sizeof(parentInfo), error = kThreadError_NotFound);
+
+        memset(&mParent, 0, sizeof(mParent));
+        memcpy(&mParent.mMacAddr, &parentInfo.mExtAddress, sizeof(mParent.mMacAddr));
+        mParent.mMode = ModeTlv::kModeFFD | ModeTlv::kModeRxOnWhenIdle | ModeTlv::kModeFullNetworkData;
```

Comment:
Updated, thanks. It seems that we also missed it here: https://github.com/openthread/openthread/blob/master/src/core/thread/mle.cpp#L2646